site/content/docs/main/restore-reference.md

```diff
@@ -36,57 +22,118 @@ Available Commands:
 ```
 
-The clusters which are not explicitly specifying nodePorts still may need to restore original NodePorts in case of disaster. Auto assigned nodePorts most probably defined on Load Balancers which located front side of cluster.  Changing all these nodePorts on Load Balancers is another operation complexity after disaster if nodePorts are changed.
 
-Velero has a flag to let user deciding the preservation of nodePorts. **`velero restore create`** sub command has  **`--preserve-nodeports`** flag to **preserve** Service nodePorts **always** regardless of nodePorts **explicitly specified** or **not**. This flag used for preserving the original nodePorts from backup and can be used as **`--preserve-nodeports`** or **`--preserve-nodeports=true`**
+## Restoring Persistent Volumes and Persistent Volume Claims
 
-If this flag given and/or set to true, Velero does not remove the nodePorts when restoring Service and tries to use the nodePorts which written on backup.
+Velero has three approaches when restoring a PV, depending on how the backup was taken.
 
-Trying to preserve nodePorts may cause **port conflicts** when restoring on situations below:
+1. When restoring a snapshot, Velero statically creates the PV and then binds it to a restored PVC. Velero's PV rename and remap process is used only in this case because this is the only case where Velero creates the PV resource directly.
+1. When restoring with Restic, Velero uses Kubernetes’ [dynamic provision process](https://kubernetes.io/docs/concepts/storage/dynamic-provisioning/) to provision the PV after creating the PVC. In this case, the PV object is not actually created by Velero.
+1. When restoring with the [CSI plugin](csi.md), the PV is created from a CSI snapshot by the CSI driver. Velero doesn’t create the PV directly, instead Velero creates a PVC with its DataSource referring to the CSI VolumeSnapshot object.
 
-- If the nodePort from the backup already allocated on the target cluster then Velero prints error log as shown below and continue to restore operation.
+### Snapshot PV Restore
 
-  ```
-  time="2020-11-23T12:58:31+03:00" level=info msg="Executing item action for services" logSource="pkg/restore/restore.go:1002" restore=velero/test-with-3-svc-20201123125825
+PV data backed up by durable snapshots is restored by VolumeSnapshot plugins. Velero calls the plugins’ interface to create a volume from a snapshot. The plugin returns the volume’s volumeID. This ID is created by storage vendors and will be updated into the PV object created by Velero, so that the PV object is connected to the volume restored from a snapshot.
```

Comment:
Says "will be updated into the PV object", I think it should be "will be updated in the PV object"